fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeStmt.java

```diff
@@ -17,19 +17,181 @@
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.UserException;
+import org.apache.doris.mysql.privilege.PaloAuth;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.qe.ConnectContext;
+
+import com.clearspring.analytics.util.Lists;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Collect statistics about a database
- *
+ * <p>
  * syntax:
  * ANALYZE [[ db_name.tb_name ] [( column_name [, ...] )], ...] [ PROPERTIES(...) ]
- *
- *     db_name.tb_name: collect table and column statistics from tb_name
- *
- *     column_name: collect column statistics from column_name
- *
- *     properties: properties of statistics jobs
- *
+ * <p>
+ * db_name.tb_name: collect table and column statistics from tb_name
+ * <p>
+ * column_name: collect column statistics from column_name
+ * <p>
+ * properties: properties of statistics jobs
  */
 public class AnalyzeStmt extends DdlStmt {
+    private final TableName dbTableName;
+    private final List<String> columnNames;
+    private Map<String, String> properties;
+
+    // after analyzed
+    private Database db;
+    private List<Table> tables;
+    private final Map<Long, List<String>> tableIdToColumnName = Maps.newHashMap();
+
+    public AnalyzeStmt(TableName dbTableName, List<String> columns, Map<String, String> properties) {
+        this.dbTableName = dbTableName;
+        this.columnNames = columns;
+        this.properties = properties;
+    }
+
+    public Database getDb() {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The db name must be obtained after the parsing is complete");
+        return this.db;
+    }
+
+    public List<Table> getTables() {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The db name must be obtained after the parsing is complete");
+        return this.tables;
+    }
+
+    public Map<Long, List<String>> getTableIdToColumnName() {
+        Preconditions.checkArgument(isAnalyzed(),
+                "The db name must be obtained after the parsing is complete");
+        return this.tableIdToColumnName;
+    }
+
+    public Map<String, String> getProperties() {
+        return this.properties;
+    }
 
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
+        super.analyze(analyzer);
+
+        // step1: analyze database and table
+        if (this.dbTableName != null) {
+            String dbName;
+            if (Strings.isNullOrEmpty(this.dbTableName.getDb())) {
+                dbName = analyzer.getDefaultDb();
+            } else {
+                dbName = ClusterNamespace.getFullName(analyzer.getClusterName(), this.dbTableName.getDb());
+            }
+
+            // check db
+            if (Strings.isNullOrEmpty(dbName)) {
```

Comment:
Please 
1. Check the valid of metadata
2. Check the permission

The logic of the two should not be interspersed and executed together

Reply:
got it.